src/server/http/version_check.js

```diff
@@ -0,0 +1,27 @@
+import { badRequest } from 'boom';
+
+export default function (kbnServer, server, config) {
+  const versionHeader = 'kbn-version';
+  const actualVersion = config.get('pkg.version');
+  const versionCheckDisabled = config.get('server.disableVersionCheck');
+  const xsrfProtectionDisabled = config.get('server.xsrf.disableProtection');
+  // this accepts the xsrf config strictly for legacy reasons
+  const disabled = versionCheckDisabled || xsrfProtectionDisabled;
```

Comment:
Maybe we should mark this with a backwards compatibility comment of some sort, maybe we should do that in more places.